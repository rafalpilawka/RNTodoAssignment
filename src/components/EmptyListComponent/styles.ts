import {StyleSheet} from 'react-native';
import normalize from 'react-native-normalize';
import {colors} from 'src/constants/platformSpecificColors';

export const styles = StyleSheet.create({
  text: {
    color: colors!.color,
    fontSize: normalize(16),
    fontWeight: 'bold',
    textAlign: 'center',
  },
  container: {
    position: 'absolute',
    width: '100%',
    height: '100%',
    justifyContent: 'center',
    alignItems: 'center',
  },
});
