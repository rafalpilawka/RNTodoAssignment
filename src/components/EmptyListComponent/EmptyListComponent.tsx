import React, {ReactElement} from 'react';
import {styles} from 'src/components/EmptyListComponent/styles';
import {Text, View} from 'react-native';

type PropsT = {
  variant: 'LISTS' | 'TODOS';
};
export const EmptyListComponent = ({variant}: PropsT): ReactElement => (
  <View style={styles.container}>
    <Text style={styles.text}>OH NO! EMPTY LIST</Text>
    <Text style={styles.text}>{`DONT HESITATE TO ADD\nSOME ${variant}`}</Text>
  </View>
);
