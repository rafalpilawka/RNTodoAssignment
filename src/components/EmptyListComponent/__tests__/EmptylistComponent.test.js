import React from 'react';
import 'react-native';
import {shallow} from 'enzyme';
import {EmptyListComponent} from 'src/components/EmptyListComponent/EmptyListComponent';
import renderer from 'react-test-renderer';

const props = {
  variant: 'LISTS',
};

describe('Empty Component', () => {
  let component;
  beforeEach(() => {
    component = shallow(<EmptyListComponent {...props} />);
    jest.clearAllMocks();
  });
  it('should match to snapshot', () => {
    const component = renderer
      .create(<EmptyListComponent {...props} />)
      .toJSON();
    expect(component).toMatchSnapshot();
  });

  it('should receive props', () => {
    const MESSAGE = `DONT HESITATE TO ADD\nSOME ${props.variant}`;
    const variantRender = component.find({children: MESSAGE});
    expect(variantRender).toHaveLength(1);
  });
});
