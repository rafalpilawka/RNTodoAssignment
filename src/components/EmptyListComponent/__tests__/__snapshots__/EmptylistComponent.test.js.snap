// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Empty Component should match to snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "height": "100%",
      "justifyContent": "center",
      "position": "absolute",
      "width": "100%",
    }
  }
>
  <Text
    style={
      Object {
        "color": Object {
          "semantic": Array [
            "systemBlue",
          ],
        },
        "fontSize": 32,
        "fontWeight": "bold",
        "textAlign": "center",
      }
    }
  >
    OH NO! EMPTY LIST
  </Text>
  <Text
    style={
      Object {
        "color": Object {
          "semantic": Array [
            "systemBlue",
          ],
        },
        "fontSize": 32,
        "fontWeight": "bold",
        "textAlign": "center",
      }
    }
  >
    DONT HESITATE TO ADD
SOME LISTS
  </Text>
</View>
`;
